sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/LogData.java

```diff
@@ -19,12 +19,6 @@
 @Immutable
 public interface LogData {
 
-  /** Returns a new {@link LogDataBuilder}. */
-  static LogDataBuilder builder(
```

Comment:
What's the motivation for this?

Reply:
It makes it more consistent with `SpanData` and doesn't couple the `LogData` to a clock impl. We anticipate that the usages are all internal anyway.

Reply:
Yeah, as Jason said, I don't know if we have any use-cases at this point for non-internal usage of this builder, so let's keep it protected in the implementation until such time as we do.

Reply:
Agree with consistency with `SpanData`.

Not sure what you mean by protected though, since the `LogDataBuilder#create()` methods are public and not in an internal package.

Reply:
ugh. perhaps it should be in an internal package? Not sure.

Reply:
The equivalent to this for tracing is `TestSpanData`. That has its own builder, and a lives in a testing package: `io.opentelemetry.sdk.testing.trace`. 

We could do the same for symmetry. 

Part of me also thinks that its unnecessarily hard to create of these `*Data` interfaces. If I want to implement a processor that modifies the `*Data` before delegating to another processor, I have to bring my own implementation or use a class with "Test" in the name (or in the case of tracing, I can subclass `DelegatingSpanData`).